NuGet.Services.Dashboard.Operations/Infrastructure/SecretConfigurationManager.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Configuration;
+using NuGet.Services.KeyVault;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace NuGet
+{
+    public static class SecretConfigurationManager
+    {
+        private static ISecretInjector _secretInjector;
+        private static ISecretReader _keyVaultReader;
+
+        public static void Init()
+        {
+            var vaultName = ConfigurationManager.AppSettings["VaultName"];
+            var clientId = ConfigurationManager.AppSettings["ClientId"];
+            var certificateThumbprint = ConfigurationManager.AppSettings["CertificateThumbprint"];
+            var validateCertificate = ConfigurationManager.AppSettings["ValidateCertificate"].Equals("true", StringComparison.InvariantCultureIgnoreCase) ? true : false;
+
+            _keyVaultReader = new KeyVaultReader(new KeyVaultConfiguration(vaultName, clientId, certificateThumbprint, validateCertificate));
+            _secretInjector = new SecretInjector(_keyVaultReader);
+
+            IEnumerable keys = new List<string>(ConfigurationManager.AppSettings.AllKeys);
+
+            foreach (string key in keys)
+            {
+                string newValue;
+                if (TryGetConfigurationValue(key, out newValue))
```

Comment:
I would prefer to leave this as is.

---
In reply to: [71935735](https://github.com/NuGet/NuGet.Services.Dashboard/pull/187#discussion_r71935735) [](ancestors = 71935735)